### PR TITLE
#47: [reminders] remind to add "web/api/..." topic labels if missing (closes #47)

### LINKED ISSUE #47: [reminders] remind to add "web/api/..." topic labels if missing
## requirements
some time after an issue is open, if it's missing one of the required labels (`web/api/escalapio/documentation`), @nemobot should make a comment to remind the creator of the issue to add one of those labels.

## specs
- create `processors/reminders.js` file
  - implement two tasks:
    -  `addReminderIfMissingTopicLabel`
      - input is GitHubEvent of a newly created issue --> if repo has at least one topic label and issue is missing a topic label launch fake event
    - `addMissingTopicLabelComment`
      - input is fake event --> GET issue --> if missing a topic label create comment and tag the creator of the issue
- topic labels:
  - `web`
  - `api`
  - `escalapio`
  - `documentation`
  - `iOS`
  - `QIA IS`
  - `QIA OD`

### PR DESCRIPTION
**Pull request by FrancescoCioria**
Issue #47

based on #48 

**test plan**
tested successfully on this issue using label `web`: https://github.com/buildo/github-prettifier/issues/51

### CODE REVIEW

File: src/processors/reminders.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { configForRepo, prettifierLog } from '../utils';
 import { some, includes } from 'lodash';
 import { isIssueEvent, isReminderEvent, isTopicReminderEvent } from '../validators';
 
-const topicLabels = ['web', 'api', 'escalapio', 'documentation'];
+const topicLabels = ['web', 'api', 'escalapio', 'documentation', 'iOS', 'QIA IS', 'QIA OD'];
```
Comment (gabro) [inline src/processors/reminders.js]: I would make this configurable, so it's more obvious where to go and change it.

File: src/processors/reminders.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  if (action === 'opened') {
+    const { github, org, name } = configForRepo(repo.name);
+    const getRepoLabels = github.repos(org, name).labels.fetch;
+
+    getRepoLabels()
+      .then(repoLabels => {
+        const repoRequiresTopicLabel = hasAtLeastOneTopicLabel(repoLabels);
+        const issueIsMissingTopicLabel = !hasAtLeastOneTopicLabel(issue.labels);
+        if (repoRequiresTopicLabel && issueIsMissingTopicLabel) {
+          onNext({ event: 'reminder-topic-label', body: { issue, repository: repo } }, 1000);
```
Comment (gabro) [inline src/processors/reminders.js]: 1s? I would make this configurable too, anyway

File: src/processors/reminders.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      });
+  }
+}
+
+export default ({ subject, onNext }) => {
+  // issues
+  subject
+    .filter(isIssueEvent)
+    .subscribe(({ body: { action, issue, repository: repo } }) => {
+      prettifierLog(`Adding reminders to issue #${issue.number} in repo ${repo.name}`);
```
Comment (gabro) [inline src/processors/reminders.js]: maybe specify which reminder in the log

File: src/processors/reminders.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      addReminderIfMissingTopicLabel(repo, issue, action, onNext);
+    });
+
+  // reminders
+  subject
+    .filter(isReminderEvent)
+    .subscribe(event => {
+      const { body: { issue, repository: repo } } = event;
+      prettifierLog(`Trigger reminders for issue #${issue.number} in repo ${repo.name}`);
```
Comment (gabro) [inline src/processors/reminders.js]: maybe specify which reminder in the log

File: src/processors/reminders.js (hunk 2 above)
Comment (gabro) [inline src/processors/reminders.js]: oh sorry, I just saw the next commit.

File: src/processors/reminders.js [hunk 5]
```diff
@@ -19,7 +19,7 @@ function addReminderIfMissingTopicLabel(repo, issue, action, onNext) {
         const repoRequiresTopicLabel = hasAtLeastOneTopicLabel(repoLabels);
         const issueIsMissingTopicLabel = !hasAtLeastOneTopicLabel(issue.labels);
         if (repoRequiresTopicLabel && issueIsMissingTopicLabel) {
-          onNext({ event: 'reminder-topic-label', body: { issue, repository: repo } }, 1000);
+          onNext({ event: 'reminder-topic-label', body: { issue, repository: repo } }, 10000);
```
Comment (gabro) [inline src/processors/reminders.js]: I would still make this configurable

**Comment by FrancescoCioria** [comment]
@gabro tested again on https://github.com/buildo/github-prettifier/issues/51

**Comment by gabro** [comment]
:+1:

**Pull request merged**